src/lib/rnp2.c

```diff
@@ -1118,6 +1070,11 @@ rnp_decrypt(rnp_ffi_t ffi, rnp_input_t input, rnp_output_t output)
         return RNP_ERROR_NULL_POINTER;
     }
 
+    // TODO: This is actually work around. Long term GH #557 needs to be implemented
```

Comment:
I can't see how this TODO makes sense at the moment. Setting the rng pointer doesn't seem like any more of a work around than setting `.password_provider` and similar fields below.

@ni4 what do you think, am I missing something? Maybe the parse handler should have an explicit `rng` field or something? (because it looks like `pgp_parse_handler_t->ctx` isn't actually used in master?)

Reply:
@dewyatt Thanks for pointing at this. Actually, being part of rnp_ctx_t is pretty enough. ctx is not used since it currently contains only options to produce data, not to parse. 
And, actually, rng logically is not needed for data decryption/signature verification - botan just use it for secert key validation. I don't think that key validation for every operation is correct, probably it would be better to move it to key provider, or other place where keys are loaded/decrypted.

Reply:
I plan to add rnp_ctx_init_ffi .